tools/buildsys/src/builder.rs

```diff
@@ -369,6 +375,37 @@ enum Retry<'a> {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// Add secrets that might be needed for builds. Since most builds won't use
+/// them, they are not automatically tracked for changes. If necessary, builds
+/// can emit the relevant cargo directives for tracking in their build script.
+fn add_secrets(args: &mut Vec<String>) -> Result<()> {
+    let sbkeys_var = "BUILDSYS_SBKEYS_PROFILE_DIR";
+    let sbkeys_dir = env::var(sbkeys_var).context(error::EnvironmentSnafu { var: sbkeys_var })?;
+
+    let sbkeys = read_dir(&sbkeys_dir).context(error::DirectoryReadSnafu { path: &sbkeys_dir })?;
+    for s in sbkeys {
+        let s = s.context(error::DirectoryReadSnafu { path: &sbkeys_dir })?;
+        args.build_secret(
+            "file",
+            &s.file_name().to_string_lossy(),
+            &s.path().to_string_lossy(),
+        );
+    }
+
+    for var in &[
+        "AWS_ACCESS_KEY_ID",
+        "AWS_SECRET_ACCESS_KEY",
+        "AWS_SESSION_TOKEN",
+    ] {
```

Comment:
How does buildsys get the region? Is it read from the `Infra.toml` file?

Reply:
The region is embedded in `kms-sign.json`, so it doesn't need to come in from the environment.